### PR TITLE
OCPBUGS-48154: Add a liveness probe to the extractor container

### PR DESCRIPTION
**Pull request by jmesnil**
The extractor container relies on critctl to collect container information

If the crio.service is restarted on the worker nodes when the CA bundle has been updated by the proxy/cluster, the container would not be able to connect anymore on the restarted CRI-O service.

Adding a liveness probe that check that crictl works as expected ensure that the container will be killed and restarted with the correct TLS settings to connect to the CRI-O service.

This fixes https://issues.redhat.com/browse/OCPBUGS-48154.

## Categories

- [X] Bugfix

## Breaking Changes

No

## References

https://issues.redhat.com/browse/OCPBUGS-48154

### CODE REVIEW

File: manifests/10-insights-runtime-extractor.yaml [hunk 1]
```diff
@@ -93,6 +93,12 @@ spec:
           env:
             - name: CONTAINER_RUNTIME_ENDPOINT
               value: unix:///crio.sock
+          livenessProbe:
+            exec:
+              command:
+                - crictl
+                - info
+              periodSeconds: 10
```
Comment (jmesnil) [inline manifests/10-insights-runtime-extractor.yaml:101]: @tremes It will take 30 seconds for the container to be restarted (`failureThreshold` is 3 by default).

Should I reduce the `failureThreshold` to 1 (minimum or 2). I don't want to decrease the `periodSeconds` as it seems overkill to check that corner case too often. wdyt?

**Comment by jmesnil** [comment]
@tremes I'm not sure but it does not look like the ci/prow/e2e-gcp-ovn-techpreview is related to the PR:

```xml
<testcase classname="tests.miscellaneous.test_cloud_token" name="test_insights_operator_disabled_no_token" time="146.360">
<failure message="wait_for.TimedOutError: Could not do 'lambda defined as `wait_for(lambda: cluster.insights_cluster_operator.is_disabled, timeout=120)`' at /insights-operator-tests/tests/miscellaneous/test_cloud_token.py:60 in time">
cluster = <ccx_ocp_testlib.cluster.Cluster object at 0x7f29af901e90> request = <FixtureRequest for <Function test_insights_operator_disabled_no_token>> @pytest.mark.techpreview def test_insights_operator_disabled_no_token(cluster: Cluster, request: FixtureRequest): """Test to check that gathering doesn't happen when the IO is disabled.""" cloud_token = cluster.pull_secret.cloud_token @request.addfinalizer def _finalize(): cluster.pull_secret.cloud_token = cloud_token cluster.restart_insights_operator() cluster.pull_secret.remove_cloud_token() cluster.restart_insights_operator() > wait_for(lambda: cluster.insights_cluster_operator.is_disabled, timeout=120) E wait_for.TimedOutError: Could not do 'lambda defined as `wait_for(lambda: cluster.insights_cluster_operator.is_disabled, timeout=120)`' at /insights-operator-tests/tests/miscellaneous/test_cloud_token.py:60 in time tests/miscellaneous/test_cloud_token.py:60: TimedOutError
</failure>
</testcase>
```

**Comment by tremes** [comment]
@jmesnil It's not. Let me retest.
/test e2e-gcp-ovn-techpreview
